src/components/TreeView/ControlledTreeView.tsx

```diff
@@ -65,7 +66,7 @@ const ControlledTreeViewForwardRef: React.ForwardRefRenderFunction<HTMLUListElem
             };
 
             const handleNodeClick = (e: React.MouseEvent) => {
-                if (onClickItem.length === 0) {
+                if (!onClickItem) {
                     toggleNode(e);
                 }
```

Comment:
That condition is redundant to the one present in the hook. The goal of the hook is to simplify implementation.